Replace n_layer with n_layers for consistency

himkt opened the pull request:
This PR replace variables/keywords `n_layer` with `n_layers`.

Although most changes do not break compatibility,
there is the breaking change in `BuildingBlock`.

https://github.com/chainer/chainer/compare/master...himkt:fix-var-name-n-layer?expand=1#diff-b27899a8e6e2ef04c4b64bb26c60669aR548

emcastillo commented on chainer/links/model/vision/resnet.py:
```diff
@@ -545,15 +545,15 @@ class BuildingBlock(link.Chain):
 
-    def __init__(self, n_layer, in_channels, mid_channels,
+    def __init__(self, n_layers, in_channels, mid_channels,
                  out_channels, stride, initialW=None, downsample_fb=False):
```
I think we should use keyword arguments to avoid breaking compatibility.

himkt commented:
I converted arguments in BuildingBlock into keyword arguments.
I updated `BuildingBlock` warns if argument `n_layer` is specified explicitly.

### Example

```python
from chainer.links.model.vision.resnet import BuildingBlock


b1 = BuildingBlock(10, 1, 1, 1)
print(b1._forward)
b2 = BuildingBlock(n_layers=10, in_channels=2, mid_channels=4, out_channels=5)
print(b2._forward)
b3 = BuildingBlock(n_layer=10, in_channels=2, mid_channels=4, out_channels=5)
print(b3._forward)
```

### Result

```
['a', 'b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8', 'b9']
['a', 'b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8', 'b9']
/Users/himkt/work/github.com/himkt/chainer/chainer/links/model/vision/resnet.py:555: UserWarning: Argument `n_layer` is deprecated. Please use `n_layers` instead
  warnings.warn('Argument `n_layer` is deprecated. '
['a', 'b1', 'b2', 'b3', 'b4', 'b5', 'b6', 'b7', 'b8', 'b9']
```

@emcastillo PTAL. :pray:

himkt commented:
Failing `pfn-ci-bot` is related to my comment. 😉

emcastillo commented on chainer/links/model/vision/resnet.py:
```diff
@@ -545,15 +546,22 @@ class BuildingBlock(link.Chain):
-                 out_channels, stride, initialW=None, downsample_fb=False):
+    def __init__(self, n_layers=None, in_channels=None, mid_channels=None,
+                 out_channels=None, stride=None, initialW=None,
+                 downsample_fb=None, n_layer=None):
```
Instead of explicitly adding a redundant parameter, code would be nicer using `**kwargs`
Thanks!

himkt replied:
I revised my PR to use `**kwargs`. :pray:

emcastillo commented on chainer/links/model/vision/resnet.py:
```diff
@@ -548,13 +548,14 @@ class BuildingBlock(link.Chain):
 
```
Can you add the n_layer to the doc string and notify that it is deprecated?
Otherwise the interface changes wont be reflected on the docs.

himkt replied:
Oh, sorry but I forgot it. I will update a document. :pray:

himkt replied:
Updated docstring in 9c7b8fe.

emcastillo replied:
Thanks!!

emcastillo commented:
Jenkins, test this please
